tree/treeplayer/src/TDFUtils.cxx

```diff
@@ -36,6 +36,79 @@ TIgnoreErrorLevelRAII::~TIgnoreErrorLevelRAII()
    gErrorIgnoreLevel = fCurIgnoreErrorLevel;
 }
 
+const std::type_info &TypeName2TypeID(const std::string &name)
+{
+   if (auto c = TClass::GetClass(name.c_str())) {
+      return *c->GetTypeInfo();
+   } else if (name == "char" || name == "Char_t")
+      return typeid(char);
+   else if (name == "unsigned char" || name == "UChar_t")
+      return typeid(unsigned char);
+   else if (name == "int" || name == "Int_t")
+      return typeid(int);
+   else if (name == "unsigned int" || name == "UInt_t")
+      return typeid(unsigned int);
+   else if (name == "short" || name == "Short_t")
+      return typeid(short);
+   else if (name == "unsigned short" || name == "UShort_t")
+      return typeid(unsigned short);
+   else if (name == "long" || name == "Long_t")
+      return typeid(long);
+   else if (name == "unsigned long" || name == "ULong_t")
+      return typeid(unsigned long);
+   else if (name == "double" || name == "Double_t")
+      return typeid(double);
+   else if (name == "float" || name == "Float_t")
+      return typeid(float);
+   else if (name == "Long64_t")
+      return typeid(Long64_t);
+   else if (name == "ULong64_t")
+      return typeid(ULong64_t);
+   else if (name == "bool" || name == "Bool_t")
+      return typeid(bool);
+   else {
+      std::string msg("Cannot extract type_info of type ");
+      msg += name.c_str();
+      msg += ".";
+      throw std::runtime_error(msg);
+   }
+}
+
+std::string TypeID2TypeName(const std::type_info &id)
```

Comment:
as above, maybe we should leave a comment for posterity explaining that we do not deal with pointer/reference types because TDF does not accept columns of pointer type and references are removed before getting here